yt_dlp/extractor/panopto.py

```diff
@@ -0,0 +1,427 @@
+                'channel_id': traverse_obj(item, ('Parent', 'Id'))
+            }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        base_url, playlist_id = mobj.group('base_url', 'id')
+
+        video_id = get_first(parse_qs(url), 'id')
+        if video_id:
+            if self.get_param('noplaylist'):
+                self.to_screen('Downloading just video %s because of --no-playlist' % video_id)
+                return self.url_result(base_url + f'/Pages/Viewer.aspx?id={video_id}', ie_key=PanoptoIE.ie_key(), video_id=video_id)
+            else:
+                self.to_screen(f'Downloading playlist {playlist_id}; add --no-playlist to just download video {video_id}')
+
+        playlist_info = self._call_api(base_url, f'/Api/Playlists/{playlist_id}', playlist_id)
+        return self.playlist_result(
+            self._entries(base_url, playlist_id, playlist_info['SessionListId']),
+            playlist_id=playlist_id, playlist_title=playlist_info.get('Name'),
+            playlist_description=playlist_info.get('Description'))
+
+
+class PanoptoListIE(PanoptoBaseIE):
+    _VALID_URL = PanoptoBaseIE.BASE_URL_RE + r'/Pages/Sessions/List\.aspx'
+    _PAGE_SIZE = 250
+    _TESTS = [
+        {
+            'url': 'https://demo.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx#folderID=%22e4c6a2fc-1214-4ca0-8fb7-aef2e29ff63a%22',
+            'info_dict': {
+                'id': 'e4c6a2fc-1214-4ca0-8fb7-aef2e29ff63a',
+                'title': 'Showcase Videos'
+            },
+            'playlist_mincount': 140
+
+        },
+        {
+            'url': 'https://demo.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx#view=2&maxResults=250',
+            'info_dict': {
+                'id': 'list',
+                'title': 'list'
+            },
+            'playlist_mincount': 300
+        },
+        {
+            # Folder that contains 8 folders and a playlist
+            'url': 'https://howtovideos.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx?noredirect=true#folderID=%224b9de7ae-0080-4158-8496-a9ba01692c2e%22',
+            'info_dict': {
+                'id': '4b9de7ae-0080-4158-8496-a9ba01692c2e',
+                'title': 'Video Tutorials'
+            },
+            'playlist_mincount': 9
+        }
+
+    ]
+
+    def _fetch_page(self, base_url, query_params, display_id, page):
+
+        params = {
+            'sortColumn': 1,
+            'getFolderData': True,
+            'includePlaylists': True,
+            **query_params,
+            'page': page,
+            'maxResults': self._PAGE_SIZE,
+        }
+
+        response = self._call_api(
+            base_url, '/Services/Data.svc/GetSessions', display_id + f' page {page+1}',
+            data={'queryParameters': params}, fatal=False)
+        if not response:
+            return  # TODO this should be fatal but being fatal makes us infinitely hit the site
```

Comment:
can u explain?

Reply:
If I set the above API call to be fatal, and we get an error (e.g HTTPError), I would expect the playlist download to stop. But something with the OnDemandPagedList would make it continuously hit the site instead? Unless I'm missing something